### PR TITLE
Document how to update dense vector field type

### PR DESCRIPTION
**Pull request by tteofili**
This documents the changes introduced in #106591 (8.15) and #110928 (8.16) to make dense vector field types updatable.

### CODE REVIEW

File: docs/reference/mapping/types/dense-vector.asciidoc [hunk 1]
```diff
@@ -448,3 +448,72 @@ POST /my-bit-vectors/_search?filter_path=hits.hits
...
+
+* updating `type` from `hnsw` to `int8_hnsw` is possible when:
+** the `m` setting is updated to the same or larger value
+** the `confidence_interval` setting is either unspecified in the original mapping or it doesn't get changed
+* updating `type` from `int8_hnsw` to `int8_hnsw` is possible when:
+** the `m` setting is updated to the same or larger value
+** the `confidence_interval` setting is either unspecified in the original mapping or it doesn't get updated
+* updating `type` from `int4_hnsw` to `int4_hnsw` is possible when:
+** the `m` setting is updated to the same or larger value
+** the `confidence_interval` setting is left untouched
```
Comment (benwtrent) [inline docs/reference/mapping/types/dense-vector.asciidoc:470]: This reads weird. Maybe just indicate that the number of connections `m` must always increase, can never decrease and for scalar quantized formats the confidence interval must be consistent.

File: docs/reference/mapping/types/dense-vector.asciidoc [hunk 2]
```diff
@@ -448,3 +448,72 @@ POST /my-bit-vectors/_search?filter_path=hits.hits
...
+                "index_options": {
+                    "type": "flat"
+                }
+            }
+        }
+    }
+}
+--------------------------------------------------
+
+Changing the `type` to `int4_hnsw` makes sure vectors indexed after that will use an int4 scalar quantized representation and HNSW (e.g., for KNN queries).
```
Comment (carlosdelest) [inline docs/reference/mapping/types/dense-vector.asciidoc:497]:
```suggestion
Changing the `type` to `int4_hnsw` makes sure vectors indexed after the change will use an int4 scalar quantized representation and HNSW (e.g., for KNN queries).
That includes mew segments created by <<index-modules-merge,merging>> previously created segments.
```

File: docs/reference/mapping/types/dense-vector.asciidoc [hunk 3]
```diff
@@ -448,3 +448,72 @@ POST /my-bit-vectors/_search?filter_path=hits.hits
...
+                "index_options": {
+                    "type": "int4_hnsw"
+                }
+            }
+        }
+    }
+}
+--------------------------------------------------
+
+Vectors indexed before this change will keep using the `flat` type (raw float32 representation and brute force search for KNN queries).
```
Comment (carlosdelest) [inline docs/reference/mapping/types/dense-vector.asciidoc:517]: Should we add some information about how to update the previous vectors to the new format, either by reindexing or force merging? 

I guess we don't want to push for force merging, but users may be interested in how to actually change the indexing type for the previuosly existing vectors...

File: docs/reference/mapping/types/dense-vector.asciidoc (hunk 1 above)
Comment (tteofili) [reply docs/reference/mapping/types/dense-vector.asciidoc:470]: thanks Ben, I like your suggestion.

File: docs/reference/mapping/types/dense-vector.asciidoc (hunk 2 above)
Comment (tteofili) [reply docs/reference/mapping/types/dense-vector.asciidoc:497]: thanks Carlos for improving it ;)

File: docs/reference/mapping/types/dense-vector.asciidoc (hunk 3 above)
Comment (tteofili) [reply docs/reference/mapping/types/dense-vector.asciidoc:507]: good point Carlos, I'll add a note about that.

**Review by benwtrent — APPROVED** [review]

**Pull request merged**